src/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -24,6 +24,58 @@ private struct DateTimeParseData
             public byte OffsetToken;
         }
 
+        public static string FormatDateTimeOffset(DateTimeOffset value)
+        {
+            Span<byte> span = stackalloc byte[JsonConstants.MaximumFormatDateTimeOffsetLength];
+
+            JsonWriterHelper.WriteDateTimeOffsetTrimmed(span, value, out int bytesWritten);
+
+            return JsonReaderHelper.GetTextFromUtf8(span.Slice(0, bytesWritten));
+        }
+
+        public static string FormatDateTime(DateTime value)
+        {
+            Span<byte> span = stackalloc byte[JsonConstants.MaximumFormatDateTimeOffsetLength];
+
+            JsonWriterHelper.WriteDateTimeTrimmed(span, value, out int bytesWritten);
+
+            return JsonReaderHelper.GetTextFromUtf8(span.Slice(0, bytesWritten));
+        }
+
+        public static bool TryParseAsISO(ReadOnlySpan<char> source, out DateTime value)
+        {
+            if (!JsonReaderHelper.IsValidDateTimeOffsetParseLength(source.Length))
+            {
+                value = default;
+                return false;
+            }
+
+            int length = JsonReaderHelper.GetUtf8ByteCount(source);
+
+            Span<byte> bytes = length <= JsonConstants.StackallocThreshold ? stackalloc byte[length] : new byte[length];
```

Comment:
`stackalloc byte[length]` => `stackalloc byte[JsonConstants.StackallocThreshold]`

Same for all other such cases.  Since we clear in the initlocals flag, it's no more expensive to use a larger size than a smaller one, and using a constant as the size makes things more efficient.